### ISSUE TITLE
MMEX crash on startup on Debian Buster

### ISSUE THREAD

**Issue by oldnomad** [moneymanagerex/moneymanagerex#3106]
MMEX version:
 - [x] 1.3.7
 - [ ] 1.3.6

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [ ] Windows
 - [ ] Mac OSX
 - [x] Linux 

# Description of the bug

I run MMEX on Debian Buster (10), installed from GitHub deb package. When I tried upgrading from 1.3.6 to 1.5.0-RC.1, it started crashing on the startup. The crash is stable, I wasn't able to get MMEX to run even once. I tried then 1.3.7 nightlies, and 1.3.7-Beta.5 works for me, but 1.3.7-Beta.6 crashes with the same stack trace.

I don't use any user themes or other modifications to MMEX. Removing the settings (so that MMEX starts without a mmb file) doesn't help.

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Run MMEX.

Expected result: MMEX runs.

Actual result: MMEX crashes.

# Additional information

Pasting only the stack trace here, full report is attached as [mmex.xml.gz](https://github.com/moneymanagerex/moneymanagerex/files/6166594/mmex.xml.gz):
```xml
  <stack>
    <frame level="0" function="mmGUIApp::ReportFatalException(wxDebugReport::Context)" offset="000000e3"/>
    <frame level="1"/>
    <frame level="2"/>
    <frame level="3" function="CreateBitmapFromRGBA(unsigned char*, int)" offset="000000f4"/>
    <frame level="4" function="buildBitmapsFromSVG(wxString, wxString)" offset="000008c7"/>
    <frame level="5" function="mmBitmap(int)" offset="000000b0"/>
    <frame level="6"/>
    <frame level="7" function="navtree_images_list()" offset="00000059"/>
    <frame level="8" function="mmGUIFrame::createControls()" offset="000001d1"/>
    <frame level="9" function="mmGUIFrame::mmGUIFrame(mmGUIApp*, wxString const&amp;, wxPoint const&amp;, wxSize const&amp;)" offset="0000034d"/>
    <frame level="10" function="OnInitImpl(mmGUIApp*)" offset="00000ec3"/>
    <frame level="11" function="mmGUIApp::OnInit()" offset="000000d8"/>
    <frame level="12" function="wxEntry(int&amp;, wchar_t**)" offset="000000a2"/>
    <frame level="13" function="main" offset="00000028"/>
    <frame level="14" function="__libc_start_main" offset="000000eb"/>
    <frame level="15" function="_start" offset="0000002a"/>
  </stack>
```

**Comment by jeancf** [comment]
I am experiencing the exact same behavior on a vanilla installation.

```
Host: desktop1 Kernel: 5.11.6-1-MANJARO x86_64 bits: 64 compiler: gcc v: 10.2.0 Desktop: KDE Plasma 5.21.3 
           tk: Qt 5.15.2 wm: kwin_x11 dm: SDDM Distro: Manjaro Linux base: Arch Linux
```
Here is the full crash log: [mmex.txt](https://github.com/moneymanagerex/moneymanagerex/files/6169575/mmex.txt)

**Comment by Heny67** [comment]
Hi!
I have a similar problem. I install the mmex_1.5.0-RC.1-1.focal_amd64.deb file, but it doesn't start after that. The error is attached. Ubuntu Mate 20.04.2 LTS (Focal Fossa) 64-bit,MATE 1.24.0

[mmex_error.txt](https://github.com/moneymanagerex/moneymanagerex/files/6172886/mmex_error.txt)

**Comment by vomikan** [comment]
It's a bit strange.
Seems render to bitmap does not working

https://github.com/moneymanagerex/moneymanagerex/blob/master/src/images_list.cpp#L332-L338

**Comment by vomikan** [comment]
I really hope this is a misprint. 
![image](https://user-images.githubusercontent.com/6836805/111863297-d603a080-896b-11eb-94b3-cf9c805ceafa.png)

**Comment by vomikan** [comment]
Can anyone check out the latest build?

**Comment by oldnomad** [comment]
I've just tried the latest nightly (from 2021-03-21 9:37 UTC+03, so it should include the latest changes), and the symptoms are the same. I get the same stack trace, and on the console I see:

```
13:03:34: Debug: 2021-03-06 - 2020-10-30
13:03:34: Debug: 2021-03-06 - 2020-05-22
13:03:34: Debug: 2021-03-06 - 2020-05-13
13:03:34: Debug: 2021-03-06 - 2020-05-05
13:03:34: Debug: 2021-03-06 - 2020-04-13
13:03:34: Debug: 2021-03-06 - 2020-04-11
13:03:34: Debug: 2021-03-06 - 2020-03-10
13:03:34: Debug: 2021-03-06 - 2019-10-19
13:03:34: Debug: 2021-03-06 - 2019-03-27
13:03:34: Debug: 2021-03-06 - 2018-05-23
13:03:34: Debug: getNewsRSS: New articles = 0
13:03:34: Debug: Scanning [/usr/share/mmex/res/] for Theme [default]
13:03:34: Debug: Found theme [colorful]
13:03:34: Debug: Found theme [simple-black]
13:03:34: Debug: Found theme [default]
Aborted
```

**Comment by vomikan** [comment]
I beleave the reason of this issue that VFS does not working on GTK and disabled 
https://github.com/moneymanagerex/moneymanagerex/blob/master/src/mmex.cpp#L264-L291

but probablu here it used
https://github.com/moneymanagerex/moneymanagerex/blob/master/src/images_list.cpp#L298

**Comment by whalley** [comment]
It's something we need to fix, but I saw in the stack trace that it was failing in `CreateBitmapFromRGBA(unsigned char*, int)` ,so I'm not sure if that is the only issue. 

We should make the fix you highlighted and get it retested. I'm hoping you have found the issue.

**Comment by vomikan** [comment]
One more place where some crutches hase been used

https://github.com/moneymanagerex/moneymanagerex/blob/master/src/util.cpp#L1140-L1180

**Comment by oldnomad** [comment]
I tried running mmex under GDB. From the disassembly it seems that `CreateBitmapFromSVGA` is being called with null `rgba`. Going one level up, `buildBitmapsFromSVG` seems to be processing SVG document `ACC_RUBIK.svg`.

**Comment by whalley** [comment]
@vomikan As you have your linux test machine.... Do you want to give this a test for the VFS fix. 
```

            // If the file does not match an icon file then just load it into VFS
            if (!iconName2enum.count(fileName))
            {
#ifndef __WXGTK__
                if (wxNOT_FOUND != filesInVFS->Index(fileName)) // If already loaded then remove and replace
                    wxMemoryFSHandler::RemoveFile(fileName);
                wxMemoryFSHandler::AddFile(fileName, buffer->GetBufferStart()
                    , buffer->GetBufferSize());
                filesInVFS->Add(fileName);
                wxLogDebug("Theme: '%s' File: '%s' has been copied to VFS", thisTheme, fileName);
#else
                const wxString repFile = mmex::getTempFolder() + fileName;
                if (!::wxFileExists(repFile)
                    || wxFileName(fileFullPath).GetModificationTime() > wxFileName(repFile).GetModificationTime())
                {
                    if (!::wxCopyFile(fileFullPath, repFile))
                        wxLogError("Could not copy %s !", fileFullPath);
                    else
                        wxLogDebug("Copying file:\n %s \nto\n %s", fileFullPath, repFile);
                }
#endif
                continue;
            }
```

**Comment by whalley** [comment]
> One more place where some crutches hase been used
> 
> https://github.com/moneymanagerex/moneymanagerex/blob/master/src/util.cpp#L1140-L1180

Yes, think that should work OK once we get the files moved to temp dir for Linux builds.

**Comment by whalley** [comment]
I'm in the process of building a Debian build/test machine...

**Comment by vomikan** [comment]
It's a trick. Change #ifndef __WXGTK__ to #ifdef __WXGTK__

**Comment by vomikan** [comment]
![image](https://user-images.githubusercontent.com/6836805/111916375-b6639980-8a8b-11eb-97ce-1663535ada51.png)

![image](https://user-images.githubusercontent.com/6836805/111916431-f460bd80-8a8b-11eb-80fc-876e5b9c62eb.png)

**Comment by whalley** [comment]
My error - I defined `fileFullPath` earlier it should be the source file path+name.  Anyway, just got it built on Debian will do some testing now.

**Comment by whalley** [comment]
Phew! Now loads icons and hopefully master.css file on Linux builds.....   Crashes when starting the reports though..... so a bit more work to do!

**Comment by whalley** [comment]
Hmmm.. The errors and crash on the reports seem very familiar to this: https://github.com/moneymanagerex/moneymanagerex/issues/1473

Sometimes the reports render and then I will get a crash. Would be good to get some feedback from Linux users to see if this is persistent.

**Comment by vomikan** [comment]
One more issue is:
https://github.com/moneymanagerex/moneymanagerex/issues/2288

**Comment by whalley** [comment]
The linux builds are still using wxWidgets 3.0.4 which is 3 years old.... I think we need to update this before progressing any further with debugging the general failures.

**Comment by vomikan** [comment]
I previously built the program from the source code.
https://github.com/moneymanagerex/moneymanagerex/issues/2472
At that time, it turned out that VFS was not working properly.

**Comment by whalley** [comment]
My view is that we close this issue as the original problem has been resolved.  We need another issue to get the linux builds up to at least wx3.1.4. I don't think we should be expending efforts further debugging until that is done and we can assess what is a wx/gtk problem and what is not. Would be good to get somebody focused on support of the Linux variants,
